OpenSim/Common/C3DFileAdapter.cpp

```diff
@@ -174,56 +188,35 @@ C3DFileAdapter::extendRead(const std::string& fileName) const {
     }
 
     if(fp_force_pts->GetItemNumber() != 0) {
-        force_table.
-            updTableMetaData().
-            setValueForKey("CalibrationMatrices", std::move(fpCalMatrices));
-
-        force_table.
-            updTableMetaData().
-            setValueForKey("Corners",             std::move(fpCorners));
-
-        force_table.
-            updTableMetaData().
-            setValueForKey("Origins",             std::move(fpOrigins));
-
-        force_table.
-            updTableMetaData().
-            setValueForKey("Types",               std::move(fpTypes));
-
-        force_table.
-            updTableMetaData().
-            setValueForKey("DataRate", 
-                           std::to_string(acquisition->GetAnalogFrequency()));
 
-        ValueArray<std::string> labels{};
+        std::vector<std::string> labels{};
         ValueArray<std::string> units{};
         for(int fp = 1; fp <= fp_force_pts->GetItemNumber(); ++fp) {
             auto fp_str = std::to_string(fp);
 
-            labels.upd().push_back(SimTK::Value<std::string>("f" + fp_str));
+            labels.push_back(SimTK::Value<std::string>("f" + fp_str));
             auto force_unit = acquisition->GetPointUnits().
                 at(_unit_index.at("force"));
             units.upd().push_back(SimTK::Value<std::string>(force_unit));
 
-            labels.upd().push_back(SimTK::Value<std::string>("m" + fp_str));
+            labels.push_back(SimTK::Value<std::string>("m" + fp_str));
             auto moment_unit = acquisition->GetPointUnits().
                 at(_unit_index.at("moment"));
             units.upd().push_back(SimTK::Value<std::string>(moment_unit));
 
-            labels.upd().push_back(SimTK::Value<std::string>("p" + fp_str));
+            labels.push_back(SimTK::Value<std::string>("p" + fp_str));
             auto position_unit = acquisition->GetPointUnits().
                 at(_unit_index.at("marker"));
             units.upd().push_back(SimTK::Value<std::string>(position_unit));
         }
-        TimeSeriesTableVec3::DependentsMetaData force_dep_metadata{};
-        force_dep_metadata.setValueArrayForKey("labels", labels);
-        force_dep_metadata.setValueArrayForKey("units", units);
-        force_table.setDependentsMetaData(force_dep_metadata);
+
+        const size_t nf = fp_force_pts->GetFrontItem()->GetFrameNumber();
+        
+        std::vector<double> force_times(nf);
+        SimTK::Matrix_<SimTK::Vec3> force_matrix(nf, labels.size());
 
         double time_step{1.0 / acquisition->GetAnalogFrequency()};
-        for(int f = 0;
-            f < fp_force_pts->GetFrontItem()->GetFrameNumber();
-            ++f) {
+        for(int f = 0; f < force_times.size();  ++f) {
```

Comment:
A straight replacement would cause a compiler warning (type mismatch). Would need to do
```cpp
for (int f = 0; f < static_cast<int>(nf); ++f)
```